### ISSUE TITLE
NODE_GYP_FORCE_PYTHON doesn't affect gyp-mac-tool

### ISSUE THREAD

**Issue by cakoose** [nodejs/node-gyp#2146]
Node 12.14.1
macOS 10.15.5
node-gyp 5.1.1

Setting the environment variable NODE_GYP_FORCE_PYTHON seems to work for most things, but it doesn't change the Python version used by "gyp-mac-tool".  I think the problem is that even though "gyp-mac-tool" is a Python program, it is just run as a generic executable with a shebang line::

```
#!/usr/bin/env python
```

So it ends up using whatever "python" is in the PATH, instead of what [`PythonFinder`](https://github.com/nodejs/node-gyp/blob/c255ffbf6adc6b60cb39fa1aa40bfc8eb45f80ac/lib/find-python.js#L17) returns.

I'm not sure how to fix this...
- Maybe run "mac-gyp-tool" as "$PYTHON mac-gyp-tool" instead of as a generic executable?  Does this require changes to gyp?
- Maybe node-gyp can do some temporary symlink/PATH stuff to ensure the right Python gets run?

**Comment by bnoordhuis** [comment]
Yes, it requires changes to gyp (well, [gyp-next](https://github.com/nodejs/gyp-next)).

Another option: start mac-gyp-tool as a shell script, then re-execute it as a python script with an appropriate interpreter; see nodejs/node@c6da5c8cdfae9e80d63f090c4289fd146c3175df for an example. That's different from inheriting `NODE_GYP_FORCE_PYTHON` at configure time, however.

I'm curious though: what issues does using a different python binary cause? Doesn't the script work with both python 2 or 3?

**Comment by cakoose** [comment]
> I'm curious though: what issues does using a different python binary cause? Doesn't the script work with both python 2 and 3?

Yes, the latest script works with both Python 2 and 3.  But I recently had to deal with a situation where a 3rd party module used node-gyp but didn't list it as a dependency, so the "install" step used up whatever is globally installed.  When I install the latest Node 12.x with nodenv, I get an older version of node-gyp that doesn't work on Python 3.

This is easy to fix on my machine, but I needed things to work on all ~50 of my coworkers machines as well.  If NODE_GYP_FORCE_PYTHON worked, that would have been an easy temporary workaround. 

Hopefully the Python 2 vs 3 thing won't be an issue anymore, but if NODE_GYP_FORCE_PYTHON affects _most_ of gyp's Python code, it's confusing when it doesn't affect "gyp-mac-tool" (or any other scripts), especially when you're trying to debug what's going on.
